org.eclipse.xtext.ui.shared/src/org/eclipse/xtext/ui/shared/internal/Activator.java

```diff
@@ -76,21 +76,25 @@ protected void initializeInjector(BundleContext context) throws CoreException {
 		Module module = new SharedModule(context);
 		if (extensions.length != 0) {
 			int numberOfMixedInModules = 0;
+			StringBuilder sb = new StringBuilder();
+			sb.append("xtext.ui.shared: Multiple contributions to extension point 'org.eclipse.xtext.ui.shared.overridingGuiceModule'."
+					+ " Will use them in unspecified order.");
 			for (IExtension iExtension : extensions) {
 				IConfigurationElement[] elements = iExtension.getConfigurationElements();
 				for (IConfigurationElement e : elements) {
 					try {
 						Module m = (Module) e.createExecutableExtension("class");
 						module = Modules.override(module).with(m);
 						numberOfMixedInModules++;
-						if (numberOfMixedInModules == 2) {
-							log.warn("Multiple overriding guice modules. Will use them in unspecified order.");
-						}
+						sb.append("\n\tcontributor: ").append(e.getContributor().getName());
 					} catch (CoreException e1) {
 						log.error(e1.getMessage(), e1);
 					}
 				}
 			}
+			if (numberOfMixedInModules >= 2) {
+				log.warn(sb.toString());
```

Comment:
It might be ok to have multiple mixins. Shall we be able to suppress this warning somehow?

Reply:
but the behaviour will be unspecified if order is random. we dont see which module overrides which

Reply:
True.
Assuming there are two bindings for the same key in the contributed modules: Likely one of the languages won't work anymore. So a warning is not sufficient in these cases.
If the bindings are not contradicting each other, the warning won't help either.

What if:
Obtain all modules at once and call `Modules.override(module).with(collectionOfAllModules);`
In case of contradictory bindings, this will blow up in the users face rather than silently with a warning.

Yes, this *might* break existing working setups but it would be a saner default, won't it?

Reply:
indeed

Reply:
In development I would prefer the fail-fast.
On the other hand, this situation might be relevant at the end user installing a bundle combination not expected.
Having it breaking is perhaps worse than hope for the best? E.g. conflict is in a minor important tweeking. But the remaining essential stuff is OK.
Then the user would be better with the current behavior.